cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/plugin/MacroParser.java

```diff
@@ -213,7 +213,7 @@ private MacroMetadata getMacroFunctionMetadata(int startIndex, int endIndex, Str
     if (functionsEnabled && (functionWhitelist.isEmpty() || functionWhitelist.contains(macroFunction))) {
       try {
         // this macro function is not returning map as a value, call evaluate()
-        if (!macroEvaluator.getMapFunctions().contains(macroFunction)) {
+        if (!macroEvaluator.evaluateAs(macroFunction).equals(MacroObjectType.MAP)) {
```

Comment:
cleaner to have a switch on the type, throwing an IllegalStateException if it is not string or map. That way it is more clear what needs to change if an enum value is added.